File: ts/project/.sdk/tm/php/core/Control.php
<?php
declare(strict_types=1);

// ProjectName SDK control

class ProjectNameControl
{
    public mixed $throw_err;
    public mixed $err;
    public mixed $explain;

    // Per-call actor, read by the `audit` and `cost` features to attribute an
    // operation to whoever asked for it.
    //
    // Declared, because this class is typed. ts carries it without declaring
    // it - its control object is open, so a caller-supplied `actor` simply
    // survives - and both features were written against that, commenting it
    // as "an optional extension property on the control". Here the property
    // did not exist, so `$ctx->ctrl->actor ?? null` was always null and every
    // operation was attributed to 'anonymous' no matter what the caller
    // passed. The shared feature corpus caught it.
    public mixed $actor;

    public function __construct(array $opts = [])
    {
        $this->throw_err = $opts['throw_err'] ?? null;
        $this->err = null;
        $this->explain = $opts['explain'] ?? null;
        $this->actor = $opts['actor'] ?? null;
    }
}

File: ts/project/.sdk/tm/rb/core/control.rb
# ProjectName SDK control

class ProjectNameControl
  attr_accessor :throw_err, :err, :explain, :actor, :paging

  def initialize(opts = {})
    @throw_err = opts[:throw_err]
    @err = nil
    @explain = opts[:explain]
    # Per-call audit actor (used by the audit feature).
    @actor = opts[:actor]
    # Per-call paging override (used by the paging feature).
    @paging = opts[:paging]
  end
end
